### PR TITLE
Fix: new() May Be Called Multiple Times

### PR DESCRIPTION
**Pull request by lempire123**
The engine initialisation function `new()` is not limited in the number of times it may be called by the owner.

**Recommendations**:

Only allow `new()` to be called once at deployment.

### CODE REVIEW

**Comment by Aurora-ClickUp** [comment]
Task linked: [CU-863g0gk60 AUR-20 new() May Be Called Multiple Times](https://app.clickup.com/t/863g0gk60)

File: engine/src/lib.rs [hunk 1]
```diff
@@ -122,12 +122,13 @@ mod contract {
     #[no_mangle]
     pub extern "C" fn new() {
         let mut io = Runtime;
-        if let Ok(state) = state::get_state(&io) {
-            require_owner_only(&state, &io.predecessor_account_id());
-        }
-
-        let args: NewCallArgs = io.read_input_borsh().sdk_unwrap();
-        state::set_state(&mut io, &args.into()).sdk_unwrap();
+        state::get_state(&io).map_or_else(
```
[the remark this answers is not in the document]
Comment (joshuajbouw) [reply engine/src/lib.rs:125]: No, can't be private at all, I don't think. I don't think the contract should call it itself and should be immediatly called after a deployment as part of a batch tx.

**Comment by birchmd** [comment]
With this change we lose the ability to modify the Engine state config after the initialization. This is probably a good thing for `chain_id`, which should be static; we already have another function to modify `owner_id` and `bridge_prover_id` is unused. But maybe we also want to add a new function to allow the owner to modify `upgrade_delay_blocks`?

**Comment by lempire123** [comment]
> With this change we lose the ability to modify the Engine state config after the initialization. This is probably a good thing for `chain_id`, which should be static; we already have another function to modify `owner_id` and `bridge_prover_id` is unused. But maybe we also want to add a new function to allow the owner to modify `upgrade_delay_blocks`?

Created new PR here, adding getter/setter methods for the `upgrade_delay_blocks` variable:
 https://github.com/aurora-is-near/aurora-engine/pull/733

**Comment by joshuajbouw** [comment]
With Michael here, we need the ability to change just the `owner_id` however as bridge prover ID isn't used and we should state migrate to remove it..

**Comment by lempire123** [comment]
> With Michael here, we need the ability to change just the `owner_id` however as bridge prover ID isn't used and we should state migrate to remove it..

So we don't need the ability to change `upgrade_delay_blocks`? The new [PR](https://github.com/aurora-is-near/aurora-engine/pull/733) adds the functions to do so, but if it's not necessary I'll take them out. @joshuajbouw @birchmd

**Pull request closed without merging**